### PR TITLE
feat(helm): support annotations for certgen resources

### LINKED ISSUE #1954: Helm Chart compatibility with ArgoCD via supporting annotations for jobs
*Description*:

Currently jobs created by the helm chart to bootstrap the gateway includes 
`helm.sh/hook: pre-install` and `ttlSecondsAfterFinished: 0`, this is a particularly bad combination for argocd that cannot recognise the completion of the pre-install hook.

```yaml
apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    helm.sh/hook: pre-install
  labels:
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: gateway-helm
    app.kubernetes.io/version: v0.5.0
    helm.sh/chart: gateway-helm-v0.5.0
  name: release-name-gateway-helm-certgen
  namespace: default
spec:
  backoffLimit: 1
  completions: 1
  parallelism: 1
  template:
    metadata:
      labels:
        app: certgen
    spec:
      containers:
      - command:
        - envoy-gateway
        - certgen
        env:
        - name: ENVOY_GATEWAY_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: KUBERNETES_CLUSTER_DOMAIN
          value: cluster.local
        image: docker.io/envoyproxy/gateway:v0.5.0
        imagePullPolicy: Always
        name: envoy-gateway-certgen
      restartPolicy: Never
      securityContext:
        runAsGroup: 65534
        runAsNonRoot: true
        runAsUser: 65534
      serviceAccountName: release-name-gateway-helm-certgen
  ttlSecondsAfterFinished: 0
```

This can be easily solved by adding some argocd annotations, e.g.:

```yaml
apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    argocd.argoproj.io/hook: PreSync
    argocd.argoproj.io/hook-delete-policy: HookSucceeded
  labels:
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: gateway-helm
    app.kubernetes.io/version: v0.5.0
    helm.sh/chart: gateway-helm-v0.5.0
  name: release-name-gateway-helm-certgen
  namespace: default
spec:
  backoffLimit: 1
  completions: 1
  parallelism: 1
  template:
    metadata:
      labels:
        app: certgen
    spec:
      containers:
      - command:
        - envoy-gateway
        - certgen
        env:
        - name: ENVOY_GATEWAY_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: KUBERNETES_CLUSTER_DOMAIN
          value: cluster.local
        image: docker.io/envoyproxy/gateway:v0.5.0
        imagePullPolicy: Always
        name: envoy-gateway-certgen
      restartPolicy: Never
      securityContext:
        runAsGroup: 65534
        runAsNonRoot: true
        runAsUser: 65534
      serviceAccountName: release-name-gateway-helm-certgen
```

Unfortunately the only way to achieve this, given that the helm chart does not support annotations, is adding kustomize to the deployment chain, which makes the flow quite unclean.

Ideally, it should be possible to pass such annotations directly to the helm chart without the need for kustomize.

*Relevant Links*:
https://github.com/argoproj/argo-cd/issues/8586

### PR DESCRIPTION
**Pull request by chicco785**
**What type of PR is this?**

feat(helm): support annotations for certgen resources

**What this PR does / why we need it**:

* support annotations for certgen resources
* support configuration for certgen job ttlSecondsAfterFinished

**Which issue(s) this PR fixes**:

Fixes #1954

### CODE REVIEW

File: charts/gateway-helm/values.tmpl.yaml [hunk 1]
```diff
@@ -55,3 +55,9 @@ envoyGatewayMetricsService:
 createNamespace: false
 
 kubernetesClusterDomain: cluster.local
+
+certgen:
+  ttlSecondsAfterFinished: 0
+  annotations:
+    job: []
```
Comment (zirain) [inline charts/gateway-helm/values.tmpl.yaml:62]: should this be a map?
and maybe use
```
job:
  annotations:
rbac:
  labels: # if needed
  annotations:
```

**Comment by Xunzhuo** [comment]
You need to run make generate to update helm docs.

File: charts/gateway-helm/templates/certgen-rbac.yaml [hunk 2]
```diff
@@ -5,8 +5,15 @@ metadata:
   namespace: '{{ .Release.Namespace }}'
   labels:
   {{- include "eg.labels" . | nindent 4 }}
+  {{- if .Values.certgen.rbac.labels }}
+  {{- toYaml .Values.certgen.rbac.labels | nindent 4 }}
+  {{- end }}
   annotations:
+  {{- if .Values.certgen.rbac.annotations }}
+    {{- toYaml .Values.certgen.rbac.annotations | nindent 4 -}}
+  {{ else }}
```
Comment (arkodg) [inline charts/gateway-helm/templates/certgen-rbac.yaml:14]: why this there a `else` here, can't we just append ?

File: charts/gateway-helm/templates/certgen-rbac.yaml [hunk 3]
```diff
@@ -34,8 +48,15 @@ metadata:
...
```
Comment (arkodg) [inline charts/gateway-helm/templates/certgen-rbac.yaml:57]: same as above

File: charts/gateway-helm/templates/certgen-rbac.yaml (hunk 2 above)
Comment (chicco785) [reply charts/gateway-helm/templates/certgen-rbac.yaml:14]: we can give a try, eventually i will open a new issue :)

**Review by zirain — APPROVED** [review]
it would be better if you can add an e2e test for this.

**Comment by chicco785** [comment]
> it would be better if you can add an e2e test for this.

if you point me to an example, I am happy to give it a try :)

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**